pkg/operator/ceph/cluster/predicate.go

```diff
@@ -131,8 +131,11 @@ func watchControllerPredicate(rookContext *clusterd.Context) predicate.Funcs {
 				}
 				diff := cmp.Diff(objOld.Spec, objNew.Spec, resourceQtyComparer)
 				if diff != "" {
-					// Set the cancellation flag to stop any ongoing orchestration
-					rookContext.RequestCancelOrchestration.Set()
+					// Set the cancellation flag to stop any ongoing orchestration only if ceph version is updated
+					if objOld.Spec.CephVersion.Image != objNew.Spec.CephVersion.Image {
+						logger.Infof("ceph image is updated in the CR %q, cancelling any ongoing orchestration", objNew.Name)
+						rookContext.RequestCancelOrchestration.Set()
```

Comment:
@BlaineEXE my assumption could be wrong here, but I think setting `RequestCancelOrchestration` only if ceph Image has changed should suffice.  What do you think?

Reply:
https://github.com/rook/rook/pull/7386/files#r599625145

Reply:
thanks for sharing that. Moving it back to WIP to look more.